### ISSUE TITLE
Suggestion to upgrade your UpdateWithNull function

### ISSUE THREAD

**Issue by khkrustev** [XiaoFaye/WooCommerce.NET#353]
I wanted to use the UpdateWithNull extension, however it was acting a bit strangely and in my opinion misleading.
First, I would like to say I am trying to update more than 8000 products on daily basis.
And I would like to minimize the amounts of calls I make to the API.
So naturally I tried following your example:

`await wc.Product.UpdateWithNull(128, new { name = "test 9", weight = "", date_on_sale_from = "", date_on_sale_to = "" });`

However, I found that the name was also passed as NULL. This is:
1. Very risky in case people (like me) do not expect it
2. I would suggest to put some comment on the function that whatever properties the object has will be NULLified.

In my case this method was golden and I wanted to tweak it just a bit to a more expected behavior in case in the future I forget how it works.
My suggestion is - whatever is null - update as null, whatever is not null update value accordingly.
In my case I have to update 5 properties and so I tweaked a bit the function:
I replaced:
`json.Append($"\"{prop.Name}\": \"\", ");`
With this:
```
var value = item.GetType().GetProperty(prop.Name).GetValue(item);
if (value == null)
   json.Append($"\"{prop.Name}\": \"\", ");
else
   json.Append($"\"{prop.Name}\": \"{value}\", ");
```
Perhaps I am using it wrongly or with the wrong intention. If so, please let me know.
### Second change
Also when trying to Update an External Product you should omit the **in_stock** property otherwise you get an exception:

> External products cannot be stock managed.

Therefore another small change would be:
```
if(prop.Name == "in_stock" 
&& item.GetType().GetProperty("type") != null 
&& item.GetType().GetProperty("type").GetValue(item).ToString() == "external")
{
      continue;
}

var value = item.GetType().GetProperty(prop.Name).GetValue(item);
if (value == null)
    json.Append($"\"{prop.Name}\": \"\", ");
else
    json.Append($"\"{prop.Name}\": \"{value}\", ");
```

**Comment by XiaoFaye** [comment]
Thanks for the suggestion.

Initially, UpdateWithNull is for setting some values to Null, I didn't think of that it might be used for updating. It should work as you suggested, I will make the changes in next release.

For the second change, I prefer not to hard code this, you should be able to make this change using jsonSerializeFilter, which is more flexible.

**Comment by XiaoFaye** [comment]
@khkrustev Hi, the UpdateWithNull function has been fixed in the latest version 0.7.7, thanks!
